libr/asm/p/asm_mips_gnu.c

```diff
@@ -43,8 +45,37 @@ static int disassemble(struct r_asm_t *a, struct r_asm_op_t *op, const ut8 *buf,
 	Offset = a->pc;
 	memcpy (bytes, buf, 4); // TODO handle thumb
 
+	if ((a->cpu != pre_cpu) && (a->features != pre_features)) {
+		free (disasm_obj.disassembler_options);
+		memset (&disasm_obj, '\0', sizeof (struct disassemble_info));
+	}	
+
 	/* prepare disassembler */
-	memset (&disasm_obj,'\0', sizeof (struct disassemble_info));
+	if (a->cpu && a->cpu != pre_cpu) {
+		if (r_str_casecmp (a->cpu, "mips64r2") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa64r2;
+		} else if (r_str_casecmp (a->cpu, "mips32r2") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa32r2;
+		} else if (r_str_casecmp (a->cpu, "mips64") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa64;
+		} else if (r_str_casecmp (a->cpu, "mips32") == 0) {
+			disasm_obj.mach = bfd_mach_mipsisa32;
+		}
+		pre_cpu = a->cpu;
+	}
+
+	if (a->features && a->features != pre_features) {
+		disasm_obj.disassembler_options = r_str_new ("");
```

Comment:
i would prefer NULL in here

Reply:
I want to create a string that can be used by r_str_append， so...

Reply:
str_newf is the same as strdup(), and the append functions accept NULL as first argument iirc

Reply:
okay~

Reply:
But I found out that I need to get from the return value of r_str_append
The code becomes longer